src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -14,13 +14,17 @@
 
 @Dao
 public class MovieSessionDaoImpl implements MovieSessionDao {
+    public static final int HOUR = 23;
+    public static final int MINUTE = 59;
+
     @Override
     public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             LocalDateTime startOfDay = date.atStartOfDay();
-            LocalDateTime endOfDay = date.atTime(23, 59);
-            Query<MovieSession> movieQuery = session.createQuery("from MovieSession ms join fetch "
-                    + "ms.movie where ms.movie.id = :id "
+            LocalDateTime endOfDay = date.atTime(HOUR, MINUTE);
+            Query<MovieSession> movieQuery = session.createQuery("from MovieSession ms "
+                    + "join fetch ms.movie "
```

Comment:
can you please answer why do we need join fetch here?

Reply:
I can't understand, where I should use join fetch, app working correct

Reply:
give some tips please

Reply:
but without join fetch working correct also

Reply:
because I didn't change fetch type on Lazy, Bohdan said it's normal